app/helpers/field_validation_helper.rb

```diff
@@ -0,0 +1,90 @@
+module FieldValidationHelper
+  def validate_mandatory_text_fields(mandatory_fields, page)
+    invalid_fields = []
+    mandatory_fields.each do |field|
+      if session[field].blank?
+        invalid_fields << { field: field.to_s,
+                            text: t("coronavirus_form.#{page}.#{field}.custom_error",
+                                    default: t("coronavirus_form.errors.missing_mandatory_text_field", field: t("coronavirus_form.#{page}.#{field}.label")).humanize) }
+      end
+    end
+    invalid_fields
+  end
+
+  def validate_date_fields(year, month, day, field)
+    return [] if year.blank? && month.blank? && day.blank?
+
+    invalid_fields = []
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_year", field: field).humanize } if year.blank?
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_month", field: field).humanize } if month.blank?
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_day", field: field).humanize } if day.blank?
+    unless(invalid_fields != [] || Date.valid_date?(year.to_i, month.to_i, day.to_i))
```

Comment:
I think it's a bit confusing using `unless` here, given one of the conditions is negated?

Would `if(invalid_fields == [] && !Date.valid_date?(year.to_i, month.to_i, day.to_i))` be equivalent?

Reply:
Agreed. This was lifted from the brexit funding code; I've removed the date validation entirely, since we've not got a requirement for date fields yet.